awslambdaric/lambda_runtime_client.py

```diff
@@ -69,9 +70,23 @@ def post_init_error(self, error_response_data):
             raise LambdaRuntimeClientError(endpoint, response.code, response_body)
 
     def wait_next_invocation(self):
-        with ThreadPoolExecutor() as e:
-            fut = e.submit(runtime_client.next)
-        response_body, headers = fut.result()
+        # Calling runtime_client.next() from a separate thread unblocks the main thread,
+        # which can then process signals.
+        if self.use_thread_for_polling_next:
+            try:
+                from concurrent.futures import ThreadPoolExecutor
```

Comment:
I'm ok with keeping it here and not in constructor. Performance overhead is negligible, maybe if one day we refactor LambdaRuntimeClient() class, we will revisit the location of lazy imports
Thank you for addressing the comments